radixdlt/src/main/java/com/radixdlt/consensus/bft/VertexStore.java

```diff
@@ -232,30 +237,49 @@ private VerifiedVertexStoreState getState() {
 		);
 	}
 
+	/**
+	 * Inserts a timeout certificate into the store.
+	 * @param timeoutCertificate the timeout certificate
+	 */
+	public void insertTimeoutCertificate(TimeoutCertificate timeoutCertificate) {
+		if (this.highestTC.isEmpty()
+				|| this.highestTC.get().getView().lt(timeoutCertificate.getView())) {
+			this.highestTC = Optional.of(timeoutCertificate);
+		}
+	}
+
+	/**
+	 * Returns the highest inserted timeout certificate.
+	 * @return the highest inserted timeout certificate
+	 */
+	public Optional<TimeoutCertificate> getHighestTimeoutCertificate() {
+		return this.highestTC;
+	}
+
 	/**
 	 * Inserts a vertex and then attempts to create the next header.
 	 * If the ledger is ahead of the vertex store then returns an empty optional
 	 * otherwise an empty optional.
 	 *
 	 * @param vertex vertex to insert
 	 */
-	public void insertVertex(VerifiedVertex vertex) {
+	public Optional<BFTInsertUpdate> insertVertex(VerifiedVertex vertex) {
```

Comment:
Should not be exposing result of method here. Need to wait for async BFTInsertUpdate event. This is done for two reasons:
1. So we can implement async I/O down the line as this method requires a persistent store on success
2. So we can run recovery deterministic tests where failure of node can be simulated to be occurring with higher granularity between the message/processing async abstraction